cascade-merge/action.yml

```diff
@@ -0,0 +1,162 @@
+        # backport target branch pattern is harcoded here as `release/`
+        BACKPORT_TARGET_BRANCH_PATTERN="release/"
+        # Remove the branch pattern prefix from the original branch name to get the original version
+        ORIGINAL_VERSION="${ORIGINAL_BRANCH#"$BACKPORT_TARGET_BRANCH_PATTERN"}"
+
+        # List and sort (in reverse, sorting in semantic version way) all the branches that match the pattern
+        RELEASE_VERSIONS=$(git branch -r | grep -oP "origin/${BACKPORT_TARGET_BRANCH_PATTERN}\\K.*" | sort -Vr)
+        TARGET_VERSIONS="${{ inputs.default-branch }}"
+        
+        # Loop through all release versions
+        for VERSION in $RELEASE_VERSIONS; do
+          echo "Comparing versions $VERSION and $ORIGINAL_VERSION"
+          compare_versions "$VERSION" "$ORIGINAL_VERSION"
+          RESULT=$(compare_versions "$VERSION" "$ORIGINAL_VERSION")
+          # If the version in the loop is greater than the original version, add it to the target versions
+          if [[ $RESULT -eq 1 ]]; then
+            TARGET_VERSIONS="$TARGET_VERSIONS $VERSION"
+          fi
+        done
+        
+        NEW_PR_URLS=""
+        
+        # Loop through all target versions
+        for TARGET_VERSION in $TARGET_VERSIONS; do
+          if [[ $TARGET_VERSION == "${{ inputs.default-branch }}" ]]; then
+            TARGET_BRANCH="${{ inputs.default-branch }}"
+          else
+            TARGET_BRANCH="$BACKPORT_TARGET_BRANCH_PATTERN$TARGET_VERSION"
+          fi
+        
+          BRANCH_NAME="$PREFIX-${ORIGINAL_BRANCH}-to-$TARGET_BRANCH"
+
+          # Get the PR number of the existing PR (if any) that has the same base and head
+          PR_NUMBER=$(gh pr list --base $TARGET_BRANCH --head $BRANCH_NAME --json number --jq '.[0].number')
+        
+          # If a PR number exists, close that PR and delete the branch        
+          if [ "$PR_NUMBER" -eq "$PR_NUMBER" ] 2>/dev/null; then
+            echo_green "closing old PR $PR_NUMBER"
+            gh pr close $PR_NUMBER --delete-branch
+            echo_green "old PR $PR_NUMBER was closed"
+          fi
+        
+          # Create a new branch, merge the original branch into it, and push it
+          git checkout -b $BRANCH_NAME
+          git merge $ORIGINAL_BRANCH
+          git push -f origin $BRANCH_NAME
+          
+          # Attempt to create a new PR and store the return status
+          echo_green "attempting to create a pull request to cascade merge from $ORIGINAL_BRANCH to $TARGET_BRANCH"
+          NEW_PR_URL=$(gh pr create --title "Cascade merge from $ORIGINAL_BRANCH to $TARGET_BRANCH" --body "Automated cascade merge" --base $TARGET_BRANCH --head $BRANCH_NAME --repo ${{ github.repository }} --draft) || PR_CREATION_STATUS=$?
+          if [[ $PR_CREATION_STATUS -ne 0 ]]; then
+              # If PR creation failed, print a message and delete the branch both locally and remotely
+              echo_green "PR creation failed. Deleting branch."
```

Comment:
This should be in red as it's an error, not in green.

Reply:
this is kinda a finalization step, just to inform the caller that the workflow runs successfully despite not being able to create the request